node/pkg/watchers/evm/connectors/polygon.go

```diff
@@ -128,6 +136,20 @@ func (c *PolygonConnector) SubscribeForBlocks(ctx context.Context, errC chan err
 				if err := c.processCheckpoint(ctx, sink, checkpoint); err != nil {
 					sub.err <- fmt.Errorf("failed to process checkpoint: %w", err)
 				}
+			case ev := <-headSink:
+				if ev == nil {
+					c.logger.Error("new latest header event is nil")
+					continue
+				}
+				if ev.Number == nil {
+					c.logger.Error("new latest header block number is nil")
+					continue
+				}
+				sink <- &NewBlock{
+					Number:   ev.Number,
+					Hash:     ev.Hash(),
```

Comment:
Do you need to error check if hash is nil?

Reply:
I don't think hash can be nil geth always allocates the bytes for it.